### PR TITLE
added SimpleNNSearcher class

### PR DESCRIPTION
**Pull request by tteofili**
the `SimpleNNSearcher` class exposes Anserini's `SimpleNearestNeighborSearcher` features in pyserini.

### CODE REVIEW

File: pyserini/search/pysearch.py [hunk 1]
```diff
@@ -252,3 +252,25 @@ def get_topics(collection_name):
         for key in topics.get(topic).keySet().toArray():
             t[topic][key] = topics.get(topic).get(key)
     return t
+
+class SimpleNNSearcher:
```
Comment (lintool) [inline pyserini/search/pysearch.py:256]: `SimpleNearestNeighborSearcher` to be consistent with Java?

Comment (tteofili) [reply pyserini/search/pysearch.py:256]: ok, makes sense

**Comment by tteofili** [comment]
I updated the class name to reflect the Anserini/Java one.

**Comment by lintool** [comment]
hey @tteofili I'm wondering if the `Results` class can be shared?

```
JResult = autoclass('io.anserini.search.SimpleSearcher$Result')
JSnnResult = autoclass('io.anserini.search.SimpleNearestNeighborSearcher$Result')
```

Yes, would require more hacking on the Java side, but if you think it's a good idea, we should take the time to do it? Since we're introducing API changes anyway?

File: pyserini/pyclass.py [hunk 2]
```diff
@@ -57,6 +57,8 @@
 JDocument = autoclass('org.apache.lucene.document.Document')
 JSearcher = autoclass('io.anserini.search.SimpleSearcher')
```
Comment (lintool) [inline pyserini/pyclass.py:58]: Should we rename to `JSimpleSearcher` and `JSimpleNearestNeighborSearcher` just to make everything consistent?

Comment (tteofili) [reply pyserini/pyclass.py]: sure, good point, done.

**Comment by tteofili** [comment]
> hey @tteofili I'm wondering if the `Results` class can be shared?
> 
> ```
> JResult = autoclass('io.anserini.search.SimpleSearcher$Result')
> JSnnResult = autoclass('io.anserini.search.SimpleNearestNeighborSearcher$Result')
> ```
> 
> Yes, would require more hacking on the Java side, but if you think it's a good idea, we should take the time to do it? Since we're introducing API changes anyway?

I would avoid sharing the same class as it wouldn't make much sense to have `result.raw` and `result.contents` for the embedding case, so I had thought it was better to have a dedicated (and simpler) object for the nearest neighbor case.

**Comment by lintool** [comment]
@tteofili sounds good, agreed.

File: pyserini/pyclass.py [hunk 3]
```diff
@@ -55,8 +55,10 @@
 ### Search
 
 JDocument = autoclass('org.apache.lucene.document.Document')
-JSearcher = autoclass('io.anserini.search.SimpleSearcher')
+JSimpleSearcher = autoclass('io.anserini.search.SimpleSearcher')
 JResult = autoclass('io.anserini.search.SimpleSearcher$Result')
```
Comment (lintool) [inline pyserini/pyclass.py:59]: In the same way, `JSimpleSearcherResult` and `JSimpleNearestNeighborSearcherResult`? I know it's a bit long...
or `JSearchResult` or `JNearestNeighborResult`?

Comment (tteofili) [reply pyserini/pyclass.py]: sure, makes sense :+1:

File: pyserini/search/pysearch.py [hunk 4]
```diff
@@ -292,3 +293,25 @@ def get_topics(collection_name):
         for key in topics.get(topic).keySet().toArray():
             t[topic][key] = topics.get(topic).get(key)
     return t
+
+class SimpleNearestNeighborSearcher:
+
+    def __init__(self, index_dir: str):
+        self.object = JSimpleNearestNeighborSearcher(JString(index_dir))
+
+    def search(self, q: str, k=10) -> List[List[JSimpleNearestNeighborSearcherResult]]:
```
Comment (lintool) [inline pyserini/search/pysearch.py:302]: Is the return type here correct?

It's dispatch to this method right? `public Result[] search(String id, int d)`
https://github.com/castorini/anserini/blob/master/src/main/java/io/anserini/search/SimpleNearestNeighborSearcher.java#L78

Comment (tteofili) [reply pyserini/search/pysearch.py]: yes, correct. I was fixing it :)

**Comment by lintool** [comment]
Would it make sense to write a test case?
Like what I do here?
https://github.com/castorini/pyserini/blob/master/tests/test_search.py#L30

I suggest we use CACM? We can store the index here also? https://github.com/castorini/anserini-data/tree/master/CACM

**Comment by tteofili** [comment]
great point, I'll add such a test case.

**Comment by tteofili** [comment]
I've added a test case that downloads a vectors index for the CACM data and runs a nearest neighbor query.

**Pull request merged**